projects/packages/my-jetpack/_inc/hooks/use-notification-watcher/use-backup-needs-attention-notice.tsx

```diff
@@ -0,0 +1,128 @@
+
+	const {
+		timezone: { offset },
+	} = getDateSettings() || { offset: '0' };
+	// Using dateI18n() to apply internationalization and formatting.
+	const backupStatusLastUpdatedDate = dateI18n(
+		'F jS, Y g:ia',
+		applyTimezone( lastUpdated, parseInt( offset ) )
+	);
+
+	const troubleshootBackupsUrl = getRedirectUrl( 'jetpack-support-troubleshooting-backup' );
+	const contactSupportUrl = getRedirectUrl( 'jetpack-support' );
+
+	const noticeTitle = __( 'Oops! We couldn’t back up your site', 'jetpack-my-jetpack' );
+
+	const onPrimaryCtaClick = useCallback( () => {
+		window.open( troubleshootBackupsUrl );
+		recordEvent( 'jetpack_my_jetpack_backup_needs_attention_notice_primary_cta_click', {
+			backup_status: status,
+		} );
+	}, [ recordEvent, status, troubleshootBackupsUrl ] );
+
+	const onSecondaryCtaClick = useCallback( () => {
+		window.open( contactSupportUrl );
+		recordEvent( 'jetpack_my_jetpack_backup_needs_attention_notice_secondary_cta_click', {
+			backup_status: status,
+		} );
+	}, [ recordEvent, status, contactSupportUrl ] );
+
+	useEffect( () => {
+		if ( ! redBubbleAlerts?.backup_failure ) {
+			return;
+		}
+
+		const noticeMessage = (
+			<Col>
+				<Text mb={ 1 }>
+					{ preventWidows(
+						sprintf(
+							// Translators: %1$s is the date the last backup took place, i.e.- "Dec 7, 2024"
+							__( 'The last backup attempted on %1$s was unsuccessful.', 'jetpack-my-jetpack' ),
+							backupStatusLastUpdatedDate
+						)
+					) }
+				</Text>
+				{ errorDescription && (
+					<Text mb={ 1 }>{ preventWidows( errorDescription as string ) }</Text>
+				) }
+				<Text mb={ 1 }>
+					{ preventWidows(
+						__(
+							'Check out our troubleshooting guide or contact your hosting provider to resolve the issue.',
+							'jetpack-my-jetpack'
+						)
+					) }
+				</Text>
+			</Col>
+		);
+
+		const noticeOptions: NoticeOptions = {
+			id: 'backup-needs-attention-notice',
+			level: 'error',
+			actions: [
+				{
+					label: __( 'Read troubleshooting guide', 'jetpack-my-jetpack' ),
+					onClick: onPrimaryCtaClick,
+					noDefaultClasses: true,
+				},
+				{
+					label: __( 'Contact support', 'jetpack-my-jetpack' ),
+					onClick: onSecondaryCtaClick,
+					isExternalLink: true,
+				},
+			],
+			priority: NOTICE_PRIORITY_HIGH,
+		};
+
+		setNotice( {
+			title: noticeTitle,
+			message: noticeMessage,
+			options: noticeOptions,
+		} );
+	}, [
+		redBubbleAlerts,
+		setNotice,
+		recordEvent,
+		onPrimaryCtaClick,
+		onSecondaryCtaClick,
+		noticeTitle,
+		backupStatusLastUpdatedDate,
+		errorDescription,
+	] );
+};
+
+export default useBackupNeedsAttentionNotice;
+
+/**
+ * Applies local timezone to date (via timezone offset)
+ * @param {string} date   - a date string with format like: 2024-12-08T14:41:45.170+00:00
+ * @param {number} offset - the timezone offset in hours, like: -3
+ *
+ * @return {Date} a JavaScript Date object
+ */
+function applyTimezone( date: string, offset: number ): Date {
```

Comment:
This could be a good util function perhaps 🤔

Reply:
Yes, good point, moved to the utils folder. 👍